<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Representing single cell data in R / Bioconductor • BiocHCADataAccess</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Representing single cell data in R / Bioconductor">
<meta property="og:description" content="BiocHCADataAccess">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">BiocHCADataAccess</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.0.3</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/A_SingleCellDataRepresentation.html">Representing single cell data in R / Bioconductor</a>
    </li>
    <li>
      <a href="../articles/B_HCADataDiscovery.html">HCA Data Discovery</a>
    </li>
    <li>
      <a href="../articles/C_HCAMatrixBrowser.html">Programmatic exploration of HCA data</a>
    </li>
    <li>
      <a href="../articles/D_HCAExplorer.html">In-depth data discovery</a>
    </li>
    <li>
      <a href="../articles/E_Directions.html">Directions and Acknowledgments</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/mtmorgan/BiocHCADataAccess">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="A_SingleCellDataRepresentation_files/header-attrs-2.5/header-attrs.js"></script><link href="A_SingleCellDataRepresentation_files/anchor-sections-1.0/anchor-sections.css" rel="stylesheet">
<script src="A_SingleCellDataRepresentation_files/anchor-sections-1.0/anchor-sections.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Representing single cell data in R / Bioconductor</h1>
            
      
      
      <div class="hidden name"><code>A_SingleCellDataRepresentation.Rmd</code></div>

    </div>

    
    
<p>Objectives:</p>
<ul>
<li>Introduce SingleCellExperiment, the central data representation for single cell data in R / Bioconductor.</li>
<li>Explore the structure of SingleCellExperiment, and the basic operations used to manipulate SingleCellExperiment.</li>
<li>Introduce downstream resources, and resources for converting bewteen other single cell data representations.</li>
</ul>
<div id="the-singlecellexperiment" class="section level1">
<h1 class="hasAnchor">
<a href="#the-singlecellexperiment" class="anchor"></a>The SingleCellExperiment</h1>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">SingleCellExperiment</span><span class="op">)</span></code></pre></div>
</div>
<div id="example-data" class="section level1">
<h1 class="hasAnchor">
<a href="#example-data" class="anchor"></a>Example data</h1>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">TENxPBMCData</span><span class="op">)</span></code></pre></div>
<ul>
<li>Another excellent resource for curated single-cell data sets – <a href="https://bioconductor.org/packages/scRNAseq">scRNAseq</a>.</li>
</ul>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">sce</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/TENxPBMCData/man/TENxPBMCData.html">TENxPBMCData</a></span><span class="op">(</span><span class="st">"pbmc4k"</span><span class="op">)</span>
<span class="co">## snapshotDate(): 2020-10-02</span>
<span class="co">## see ?TENxPBMCData and browseVignettes('TENxPBMCData') for documentation</span>
<span class="co">## loading from cache</span>
<span class="va">sce</span>
<span class="co">## class: SingleCellExperiment </span>
<span class="co">## dim: 33694 4340 </span>
<span class="co">## metadata(0):</span>
<span class="co">## assays(1): counts</span>
<span class="co">## rownames(33694): ENSG00000243485 ENSG00000237613 ... ENSG00000277475</span>
<span class="co">##   ENSG00000268674</span>
<span class="co">## rowData names(3): ENSEMBL_ID Symbol_TENx Symbol</span>
<span class="co">## colnames: NULL</span>
<span class="co">## colData names(11): Sample Barcode ... Individual Date_published</span>
<span class="co">## reducedDimNames(0):</span>
<span class="co">## altExpNames(0):</span></code></pre></div>
<p><img src="images/sce-1.png" width="530"></p>
</div>
<div id="components-of-the-singlecellexperiment" class="section level1">
<h1 class="hasAnchor">
<a href="#components-of-the-singlecellexperiment" class="anchor"></a>Components of the SingleCellExperiment</h1>
<p>The ‘counts’ matrix</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/SingleCellExperiment/man/assays.html">counts</a></span><span class="op">(</span><span class="va">sce</span><span class="op">)</span>
<span class="co">## &lt;33694 x 4340&gt; matrix of class DelayedMatrix and type "integer":</span>
<span class="co">##                    [,1]    [,2]    [,3]    [,4] ... [,4337] [,4338] [,4339]</span>
<span class="co">## ENSG00000243485       0       0       0       0   .       0       0       0</span>
<span class="co">## ENSG00000237613       0       0       0       0   .       0       0       0</span>
<span class="co">## ENSG00000186092       0       0       0       0   .       0       0       0</span>
<span class="co">## ENSG00000238009       0       0       0       0   .       0       0       0</span>
<span class="co">## ENSG00000239945       0       0       0       0   .       0       0       0</span>
<span class="co">##             ...       .       .       .       .   .       .       .       .</span>
<span class="co">## ENSG00000277856       0       0       0       0   .       0       0       0</span>
<span class="co">## ENSG00000275063       0       0       0       0   .       0       0       0</span>
<span class="co">## ENSG00000271254       0       0       0       0   .       0       0       0</span>
<span class="co">## ENSG00000277475       0       0       0       0   .       0       0       0</span>
<span class="co">## ENSG00000268674       0       0       0       0   .       0       0       0</span>
<span class="co">##                 [,4340]</span>
<span class="co">## ENSG00000243485       0</span>
<span class="co">## ENSG00000237613       0</span>
<span class="co">## ENSG00000186092       0</span>
<span class="co">## ENSG00000238009       0</span>
<span class="co">## ENSG00000239945       0</span>
<span class="co">##             ...       .</span>
<span class="co">## ENSG00000277856       0</span>
<span class="co">## ENSG00000275063       0</span>
<span class="co">## ENSG00000271254       0</span>
<span class="co">## ENSG00000277475       0</span>
<span class="co">## ENSG00000268674       0</span></code></pre></div>
<ul>
<li>Gene (rows) by cell (columns) matrix of raw counts.</li>
<li>‘DelayedMatrix’: on-disk representation, so very large data sets can be represented and manipulated.</li>
</ul>
<p>Essential manipulations</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/mean.html">mean</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SingleCellExperiment/man/assays.html">counts</a></span><span class="op">(</span><span class="va">sce</span><span class="op">)</span> <span class="op">==</span> <span class="fl">0</span><span class="op">)</span>            <span class="co"># proportion of zeros</span>
<span class="co">## [1] 0.9608314</span></code></pre></div>
<ul>
<li>Proportion of 0’s? Access the counts, determine whether each entry in the matrix is a ‘zero’, and calculate the mean.</li>
</ul>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/graphics/hist.html">hist</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html">log10</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colSums.html">colSums</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SingleCellExperiment/man/assays.html">counts</a></span><span class="op">(</span><span class="va">sce</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="co"># 'library size'</span></code></pre></div>
<p><img src="A_SingleCellDataRepresentation_files/figure-html/unnamed-chunk-7-1.png" width="700"></p>
<ul>
<li>Distribution of ‘library sizes’? Sum each column and draw a histogram, using <code><a href="https://rdrr.io/r/base/Log.html">log10()</a></code> transformed values.</li>
</ul>
<p>Annotation</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="fu">rowData</span><span class="op">(</span><span class="va">sce</span><span class="op">)</span>, <span class="fl">3</span><span class="op">)</span>
<span class="co">## DataFrame with 3 rows and 3 columns</span>
<span class="co">##                      ENSEMBL_ID  Symbol_TENx      Symbol</span>
<span class="co">##                     &lt;character&gt;  &lt;character&gt; &lt;character&gt;</span>
<span class="co">## ENSG00000243485 ENSG00000243485 RP11-34P13.3          NA</span>
<span class="co">## ENSG00000237613 ENSG00000237613      FAM138A     FAM138A</span>
<span class="co">## ENSG00000186092 ENSG00000186092        OR4F5       OR4F5</span></code></pre></div>
<ul>
<li>Rich possibilities for gene and cell annotation</li>
</ul>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org">dplyr</a></span><span class="op">)</span>
<span class="fu">colData</span><span class="op">(</span><span class="va">sce</span><span class="op">)</span> <span class="op">%&gt;%</span>
    <span class="fu"><a href="https://tibble.tidyverse.org/reference/as_tibble.html">as_tibble</a></span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span>
    <span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="fl">3</span><span class="op">)</span>
<span class="co">## # A tibble: 3 x 11</span>
<span class="co">##   Sample Barcode Sequence Library Cell_ranger_ver… Tissue_status Barcode_type</span>
<span class="co">##   &lt;chr&gt;  &lt;chr&gt;   &lt;chr&gt;      &lt;int&gt; &lt;chr&gt;            &lt;chr&gt;         &lt;chr&gt;       </span>
<span class="co">## 1 pbmc4k AAACCT… AAACCTG…       1 v2.1             &lt;NA&gt;          Chromium    </span>
<span class="co">## 2 pbmc4k AAACCT… AAACCTG…       1 v2.1             &lt;NA&gt;          Chromium    </span>
<span class="co">## 3 pbmc4k AAACCT… AAACCTG…       1 v2.1             &lt;NA&gt;          Chromium    </span>
<span class="co">## # … with 4 more variables: Chemistry &lt;chr&gt;, Sequence_platform &lt;chr&gt;,</span>
<span class="co">## #   Individual &lt;chr&gt;, Date_published &lt;chr&gt;</span></code></pre></div>
<ul>
<li>Can play well enough with other analysis paradigms in R, e.g., tidyverse use of pipes and tibbles.</li>
</ul>
<p>Subsetting</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ridx</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colSums.html">rowSums</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SingleCellExperiment/man/assays.html">counts</a></span><span class="op">(</span><span class="va">sce</span><span class="op">)</span><span class="op">)</span> <span class="op">&gt;</span> <span class="fl">0</span> <span class="co"># rows with non-zero counts</span>
<span class="va">sce</span><span class="op">[</span><span class="va">ridx</span>,<span class="op">]</span>                       <span class="co"># select rows, all columns</span>
<span class="co">## class: SingleCellExperiment </span>
<span class="co">## dim: 19773 4340 </span>
<span class="co">## metadata(0):</span>
<span class="co">## assays(1): counts</span>
<span class="co">## rownames(19773): ENSG00000238009 ENSG00000239945 ... ENSG00000276345</span>
<span class="co">##   ENSG00000271254</span>
<span class="co">## rowData names(3): ENSEMBL_ID Symbol_TENx Symbol</span>
<span class="co">## colnames: NULL</span>
<span class="co">## colData names(11): Sample Barcode ... Individual Date_published</span>
<span class="co">## reducedDimNames(0):</span>
<span class="co">## altExpNames(0):</span></code></pre></div>
<ul>
<li>Coordinated row- and column- subsetting, using a ‘matrix’-like metaphore.</li>
<li>For example, selecting only rows with non-zero counts in at least one cell.</li>
</ul>
<p>Additional information in a <code>SingleCellExperiment</code></p>
<ul>
<li>Reduced dimension calculations</li>
</ul>
</div>
<div id="what-do-do-with-a-singlecellexperiment" class="section level1">
<h1 class="hasAnchor">
<a href="#what-do-do-with-a-singlecellexperiment" class="anchor"></a>What do do with a SingleCellExperiment?</h1>
<ul>
<li>Check out ‘<a href="https://bioconductor.org/books/3.12/OSCA">Orchestrating Single Cell Analysis with Bioconductor</a>’!</li>
</ul>
</div>
<div id="other-representations" class="section level1">
<h1 class="hasAnchor">
<a href="#other-representations" class="anchor"></a>Other representations</h1>
<ul>
<li>
<a href="https://bioconductor.org/packages/zelkonverter">zelkonverter</a> provides methods to convert between SingleCellExperiment and Python’s AnnData representation.</li>
<li>Converters bewtween R’s [Seurat][] package are also available.</li>
</ul>
</div>
<div id="summary" class="section level1">
<h1 class="hasAnchor">
<a href="#summary" class="anchor"></a>Summary</h1>
<ul>
<li>The [SingleCellExperiment][] is a central object for representing single cell experiments (including very large experiments) in R / Bioconductor.</li>
<li>SingleCellExperiment combines count matrices with annotations on rows and columns, as well as derived measures such as dimensionality reductions.</li>
<li>SingleCellExperiment can play well enough with other paradigms in R, e.g., the <code>tibble</code> and tidyverse.</li>
<li>‘<a href="https://bioconductor.org/books/3.12/OSCA">Orchestrating Single Cell Analysis with Bioconductor</a>’ provides an amazing resource for opportunities for down-stream analysis.</li>
<li>The SingleCellExperiment can be converted ot other popular formats, for instance Python’s AnnData.</li>
</ul>
<div id="session-info" class="section level2">
<h2 class="hasAnchor">
<a href="#session-info" class="anchor"></a>Session info</h2>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html">sessionInfo</a></span><span class="op">(</span><span class="op">)</span>
<span class="co">## R version 4.0.3 Patched (2020-10-13 r79345)</span>
<span class="co">## Platform: x86_64-apple-darwin17.7.0 (64-bit)</span>
<span class="co">## Running under: macOS High Sierra 10.13.6</span>
<span class="co">## </span>
<span class="co">## Matrix products: default</span>
<span class="co">## BLAS:   /Users/ma38727/bin/R-4-0-branch/lib/libRblas.dylib</span>
<span class="co">## LAPACK: /Users/ma38727/bin/R-4-0-branch/lib/libRlapack.dylib</span>
<span class="co">## </span>
<span class="co">## locale:</span>
<span class="co">## [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8</span>
<span class="co">## </span>
<span class="co">## attached base packages:</span>
<span class="co">## [1] stats4    parallel  stats     graphics  grDevices utils     datasets </span>
<span class="co">## [8] methods   base     </span>
<span class="co">## </span>
<span class="co">## other attached packages:</span>
<span class="co">##  [1] dplyr_1.0.2                 TENxPBMCData_1.8.0         </span>
<span class="co">##  [3] HDF5Array_1.18.0            rhdf5_2.34.0               </span>
<span class="co">##  [5] DelayedArray_0.16.0         Matrix_1.2-18              </span>
<span class="co">##  [7] SingleCellExperiment_1.12.0 SummarizedExperiment_1.20.0</span>
<span class="co">##  [9] Biobase_2.50.0              GenomicRanges_1.42.0       </span>
<span class="co">## [11] GenomeInfoDb_1.26.0         IRanges_2.24.0             </span>
<span class="co">## [13] S4Vectors_0.28.0            BiocGenerics_0.36.0        </span>
<span class="co">## [15] MatrixGenerics_1.2.0        matrixStats_0.57.0         </span>
<span class="co">## </span>
<span class="co">## loaded via a namespace (and not attached):</span>
<span class="co">##  [1] httr_1.4.2                    bit64_4.0.5                  </span>
<span class="co">##  [3] AnnotationHub_2.22.0          shiny_1.5.0                  </span>
<span class="co">##  [5] assertthat_0.2.1              interactiveDisplayBase_1.28.0</span>
<span class="co">##  [7] BiocManager_1.30.10.5         BiocFileCache_1.14.0         </span>
<span class="co">##  [9] blob_1.2.1                    GenomeInfoDbData_1.2.4       </span>
<span class="co">## [11] yaml_2.2.1                    BiocVersion_3.12.0           </span>
<span class="co">## [13] pillar_1.4.6                  RSQLite_2.2.1                </span>
<span class="co">## [15] backports_1.2.0               lattice_0.20-41              </span>
<span class="co">## [17] glue_1.4.2                    digest_0.6.27                </span>
<span class="co">## [19] promises_1.1.1                XVector_0.30.0               </span>
<span class="co">## [21] htmltools_0.5.0               httpuv_1.5.4                 </span>
<span class="co">## [23] pkgconfig_2.0.3               zlibbioc_1.36.0              </span>
<span class="co">## [25] xtable_1.8-4                  purrr_0.3.4                  </span>
<span class="co">## [27] later_1.1.0.1                 tibble_3.0.4                 </span>
<span class="co">## [29] generics_0.1.0                ellipsis_0.3.1               </span>
<span class="co">## [31] withr_2.3.0                   cli_2.1.0                    </span>
<span class="co">## [33] mime_0.9                      magrittr_1.5                 </span>
<span class="co">## [35] crayon_1.3.4                  memoise_1.1.0                </span>
<span class="co">## [37] evaluate_0.14                 fansi_0.4.1                  </span>
<span class="co">## [39] fs_1.5.0                      textshaping_0.1.2            </span>
<span class="co">## [41] tools_4.0.3                   lifecycle_0.2.0              </span>
<span class="co">## [43] stringr_1.4.0                 Rhdf5lib_1.12.0              </span>
<span class="co">## [45] AnnotationDbi_1.52.0          compiler_4.0.3               </span>
<span class="co">## [47] pkgdown_1.6.1                 systemfonts_0.3.2            </span>
<span class="co">## [49] rlang_0.4.8                   grid_4.0.3                   </span>
<span class="co">## [51] RCurl_1.98-1.2                rhdf5filters_1.2.0           </span>
<span class="co">## [53] rappdirs_0.3.1                bitops_1.0-6                 </span>
<span class="co">## [55] rmarkdown_2.5                 ExperimentHub_1.16.0         </span>
<span class="co">## [57] codetools_0.2-18              DBI_1.1.0                    </span>
<span class="co">## [59] curl_4.3                      R6_2.5.0                     </span>
<span class="co">## [61] knitr_1.30                    utf8_1.1.4                   </span>
<span class="co">## [63] fastmap_1.0.1                 bit_4.0.4                    </span>
<span class="co">## [65] rprojroot_1.3-2               ragg_0.4.0                   </span>
<span class="co">## [67] desc_1.2.0                    stringi_1.5.3                </span>
<span class="co">## [69] Rcpp_1.0.5                    png_0.1-7                    </span>
<span class="co">## [71] vctrs_0.3.4                   dbplyr_2.0.0                 </span>
<span class="co">## [73] tidyselect_1.1.0              xfun_0.19</span></code></pre></div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Martin Morgan.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
