<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>HCA Data Discovery • BiocHCADataAccess</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="HCA Data Discovery">
<meta property="og:description" content="BiocHCADataAccess">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">BiocHCADataAccess</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.0.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/A_SingleCellDataRepresentation.html">Representing single cell data in R / Bioconductor</a>
    </li>
    <li>
      <a href="../articles/B_HCADataDiscovery.html">HCA Data Discovery</a>
    </li>
    <li>
      <a href="../articles/C_HCAMatrixBrowser.html">Programmatic exploration of HCA data</a>
    </li>
    <li>
      <a href="../articles/D_HCAExplorer.html">Even more in-depth data discovery</a>
    </li>
    <li>
      <a href="../articles/E_Directions.html">Directions</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/mtmorgan/BiocHCADataAccess">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="B_HCADataDiscovery_files/header-attrs-2.5/header-attrs.js"></script><link href="B_HCADataDiscovery_files/anchor-sections-1.0/anchor-sections.css" rel="stylesheet">
<script src="B_HCADataDiscovery_files/anchor-sections-1.0/anchor-sections.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>HCA Data Discovery</h1>
            
      
      
      <div class="hidden name"><code>B_HCADataDiscovery.Rmd</code></div>

    </div>

    
    
<p>Objectives:</p>
<ul>
<li>Explore the HCA data portal via web browser.</li>
<li>Download and import an example data matrix.</li>
<li>Illustrate essential operations on single cell data in R / Bioconductor.</li>
</ul>
<div id="the-hca-data-portal" class="section level1">
<h1 class="hasAnchor">
<a href="#the-hca-data-portal" class="anchor"></a>The HCA Data Portal</h1>
<p>The HCA <a href="https://data.humancellatlas.org/explore">data portal</a></p>
<ul>
<li><p><a href="https://data.humancellatlas.org/explore" class="uri">https://data.humancellatlas.org/explore</a></p></li>
<li><p>An example (chosen arbitrarily!): <a href="https://data.humancellatlas.org/explore/projects/005d611a-14d5-4fbf-846e-571a1f874f70">Assessing the relevance of organoids to model inter-individual variation</a>.</p></li>
</ul>
<p>Different ways to export data.</p>
<ul>
<li>csv, mtx, loom.</li>
<li>Loom format is richest, with abundant metadata about cells and genes.</li>
</ul>
</div>
<div id="data-retrieval-and-import" class="section level1">
<h1 class="hasAnchor">
<a href="#data-retrieval-and-import" class="anchor"></a>Data retrieval and import</h1>
<p>Retrieve the data</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">url</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span>
    <span class="st">"https://data.humancellatlas.org/project-assets/project-matrices/"</span>,
    <span class="st">"005d611a-14d5-4fbf-846e-571a1f874f70.homo_sapiens.loom"</span>
<span class="op">)</span>
<span class="va">path</span> <span class="op">&lt;-</span> <span class="fu">BiocFileCache</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocFileCache/man/BiocFileCache-class.html">bfcrpath</a></span><span class="op">(</span>rnames <span class="op">=</span> <span class="va">url</span><span class="op">)</span></code></pre></div>
<ul>
<li>Use <a href="https://bioconductor.org/packages/BiocFileCache">BiocFileCache</a> to retrieve once, for fast easy access the next time.</li>
</ul>
<p>Load the data using <a href="https://bioconductor.org/packages/LoomExperiment">LoomExperiment</a></p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">LoomExperiment</span><span class="op">)</span>
<span class="va">loom</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/LoomExperiment/man/import-method.html">import</a></span><span class="op">(</span><span class="va">path</span><span class="op">)</span>
<span class="va">loom</span>
<span class="co">## class: LoomExperiment </span>
<span class="co">## dim: 58347 156936 </span>
<span class="co">## metadata(0):</span>
<span class="co">## assays(1): matrix</span>
<span class="co">## rownames: NULL</span>
<span class="co">## rowData names(9): Accession Gene ... genus_species isgene</span>
<span class="co">## colnames: NULL</span>
<span class="co">## colData names(44): CellID analysis_protocol.protocol_core.protocol_id</span>
<span class="co">##   ... specimen_from_organism.provenance.document_id total_umis</span>
<span class="co">## rowGraphs(0): NULL</span>
<span class="co">## colGraphs(0): NULL</span></code></pre></div>
<ul>
<li>A <code>LoomExperiment</code> is much like a <code>SummarizedExperiment</code>.</li>
<li>Transforms internal file-based representation of native <code>loom</code> format to familiar R / Bioconductor structures.</li>
</ul>
</div>
<div id="exploring-single-cell-data" class="section level1">
<h1 class="hasAnchor">
<a href="#exploring-single-cell-data" class="anchor"></a>Exploring single cell data</h1>
<p>Easy to sample and explore</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">x0</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html">as.matrix</a></span><span class="op">(</span><span class="fu">assay</span><span class="op">(</span><span class="va">loom</span><span class="op">)</span><span class="op">[</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">1000</span><span class="op">]</span><span class="op">)</span>
<span class="va">library_size</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colSums.html">colSums</a></span><span class="op">(</span><span class="va">x0</span><span class="op">)</span>
<span class="va">non_zero_genes</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colSums.html">colSums</a></span><span class="op">(</span><span class="va">x0</span> <span class="op">&gt;</span> <span class="fl">0</span><span class="op">)</span></code></pre></div>
<ul>
<li>Create a subset of data and represent as in-memory matrix.</li>
<li>‘Library size’ as sum of counts in each column.</li>
<li>How many genes with non-zero expression in each cell? Determine which counts are greater than 0 (<code>TRUE</code> or <code>FALSE</code>), sum the number of <code>TRUE</code> in each column.</li>
</ul>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html">plot</a></span><span class="op">(</span><span class="va">library_size</span> <span class="op">~</span> <span class="va">non_zero_genes</span>, log <span class="op">=</span> <span class="st">"xy"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/abline.html">abline</a></span><span class="op">(</span>v <span class="op">=</span> <span class="fl">500</span><span class="op">)</span></code></pre></div>
<p><img src="B_HCADataDiscovery_files/figure-html/unnamed-chunk-5-1.png" width="700"></p>
<ul>
<li>Bimodal distribution of cells make us appreciate need for quality assessment, etc.</li>
</ul>
<p>Extensive row and column data annotations</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="fu">rowData</span><span class="op">(</span><span class="va">loom</span><span class="op">)</span>, <span class="fl">3</span><span class="op">)</span>
<span class="co">## DataFrame with 3 rows and 9 columns</span>
<span class="co">##         Accession        Gene  chromosome featureend      featurekey</span>
<span class="co">##       &lt;character&gt; &lt;character&gt; &lt;character&gt;  &lt;integer&gt;     &lt;character&gt;</span>
<span class="co">## 1 ENSG00000000003      TSPAN6        chrX  100639991 ENSG00000000003</span>
<span class="co">## 2 ENSG00000000005        TNMD        chrX  100599885 ENSG00000000005</span>
<span class="co">## 3 ENSG00000000419        DPM1       chr20   50958555 ENSG00000000419</span>
<span class="co">##   featurestart    featuretype genus_species isgene</span>
<span class="co">##      &lt;integer&gt;    &lt;character&gt;   &lt;character&gt;  &lt;raw&gt;</span>
<span class="co">## 1    100627109 protein_coding  Homo sapiens     01</span>
<span class="co">## 2    100584802 protein_coding  Homo sapiens     01</span>
<span class="co">## 3     50934867 protein_coding  Homo sapiens     01</span></code></pre></div>
<ul>
<li>Annotations on genes from the HCA pipeline used in pre-processing.</li>
</ul>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org">dplyr</a></span><span class="op">)</span>
<span class="fu">colData</span><span class="op">(</span><span class="va">loom</span><span class="op">)</span> <span class="op">%&gt;%</span>
    <span class="fu"><a href="https://tibble.tidyverse.org/reference/as_tibble.html">as_tibble</a></span><span class="op">(</span><span class="op">)</span>
<span class="co">## # A tibble: 156,936 x 44</span>
<span class="co">##    CellID analysis_protoc… analysis_protoc… analysis_workin… barcode bundle_uuid</span>
<span class="co">##    &lt;chr&gt;  &lt;chr&gt;            &lt;chr&gt;            &lt;chr&gt;            &lt;chr&gt;   &lt;chr&gt;      </span>
<span class="co">##  1 0002f… optimus_v1.3.1   ca93a946-4c58-4… blessed          CAACCT… 4c0fb68f-c…</span>
<span class="co">##  2 00032… optimus_v1.3.1   a34b5f5b-c6de-4… blessed          TTTGTC… d9313f14-d…</span>
<span class="co">##  3 0003c… optimus_v1.3.1   c1f32245-b732-4… blessed          CCTTTC… fd6fc9ba-2…</span>
<span class="co">##  4 00053… optimus_v1.3.1   a34b5f5b-c6de-4… blessed          ATAGAC… d9313f14-d…</span>
<span class="co">##  5 0006d… optimus_v1.3.2   83af655a-7104-4… blessed          CATTAT… 74804ad7-3…</span>
<span class="co">##  6 00072… optimus_v1.3.1   a34b5f5b-c6de-4… blessed          CCTATT… d9313f14-d…</span>
<span class="co">##  7 000d3… optimus_v1.3.1   c5c73a09-8c0c-4… blessed          CTACAT… c7b2a2c7-9…</span>
<span class="co">##  8 000f4… optimus_v1.3.1   c1f32245-b732-4… blessed          AACTCC… fd6fc9ba-2…</span>
<span class="co">##  9 00107… optimus_v1.3.1   c5c73a09-8c0c-4… blessed          CTTCTC… c7b2a2c7-9…</span>
<span class="co">## 10 0020a… optimus_v1.3.1   c1f32245-b732-4… blessed          ATTTCT… fd6fc9ba-2…</span>
<span class="co">## # … with 156,926 more rows, and 38 more variables: bundle_version &lt;chr&gt;,</span>
<span class="co">## #   cell_suspension.genus_species.ontology &lt;chr&gt;,</span>
<span class="co">## #   cell_suspension.genus_species.ontology_label &lt;chr&gt;,</span>
<span class="co">## #   cell_suspension.provenance.document_id &lt;chr&gt;, derived_organ_label &lt;chr&gt;,</span>
<span class="co">## #   derived_organ_ontology &lt;chr&gt;, derived_organ_parts_label &lt;chr&gt;,</span>
<span class="co">## #   derived_organ_parts_ontology &lt;chr&gt;,</span>
<span class="co">## #   donor_organism.development_stage.ontology &lt;chr&gt;,</span>
<span class="co">## #   donor_organism.development_stage.ontology_label &lt;chr&gt;,</span>
<span class="co">## #   donor_organism.diseases.ontology &lt;chr&gt;,</span>
<span class="co">## #   donor_organism.diseases.ontology_label &lt;chr&gt;,</span>
<span class="co">## #   donor_organism.human_specific.ethnicity.ontology &lt;chr&gt;,</span>
<span class="co">## #   donor_organism.human_specific.ethnicity.ontology_label &lt;chr&gt;,</span>
<span class="co">## #   donor_organism.is_living &lt;chr&gt;,</span>
<span class="co">## #   donor_organism.provenance.document_id &lt;chr&gt;, donor_organism.sex &lt;chr&gt;,</span>
<span class="co">## #   dss_bundle_fqid &lt;chr&gt;, emptydrops_is_cell &lt;chr&gt;, file_uuid &lt;chr&gt;,</span>
<span class="co">## #   file_version &lt;chr&gt;, genes_detected &lt;int&gt;,</span>
<span class="co">## #   library_preparation_protocol.end_bias &lt;chr&gt;,</span>
<span class="co">## #   library_preparation_protocol.input_nucleic_acid_molecule.ontology &lt;chr&gt;,</span>
<span class="co">## #   library_preparation_protocol.input_nucleic_acid_molecule.ontology_label &lt;chr&gt;,</span>
<span class="co">## #   library_preparation_protocol.library_construction_method.ontology &lt;chr&gt;,</span>
<span class="co">## #   library_preparation_protocol.library_construction_method.ontology_label &lt;chr&gt;,</span>
<span class="co">## #   library_preparation_protocol.provenance.document_id &lt;chr&gt;,</span>
<span class="co">## #   library_preparation_protocol.strand &lt;chr&gt;,</span>
<span class="co">## #   project.project_core.project_short_name &lt;chr&gt;,</span>
<span class="co">## #   project.project_core.project_title &lt;chr&gt;,</span>
<span class="co">## #   project.provenance.document_id &lt;chr&gt;,</span>
<span class="co">## #   specimen_from_organism.organ.ontology &lt;chr&gt;,</span>
<span class="co">## #   specimen_from_organism.organ.ontology_label &lt;chr&gt;,</span>
<span class="co">## #   specimen_from_organism.organ_parts.ontology &lt;chr&gt;,</span>
<span class="co">## #   specimen_from_organism.organ_parts.ontology_label &lt;chr&gt;,</span>
<span class="co">## #   specimen_from_organism.provenance.document_id &lt;chr&gt;, total_umis &lt;dbl&gt;</span></code></pre></div>
<ul>
<li>Very rich data cell-level data summarized by the HCA</li>
<li>Can be linked to even more annotation information!</li>
</ul>
<p>Easily coerce to standard <code>SingleCellExperiment</code>…</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">sce</span> <span class="op">&lt;-</span> <span class="fu">as</span><span class="op">(</span><span class="va">loom</span>, <span class="st">"SingleCellExperiment"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="fu">assays</span><span class="op">(</span><span class="va">sce</span><span class="op">)</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="st">"counts"</span>
<span class="va">sce</span>
<span class="co">## class: SingleCellExperiment </span>
<span class="co">## dim: 58347 156936 </span>
<span class="co">## metadata(0):</span>
<span class="co">## assays(1): counts</span>
<span class="co">## rownames: NULL</span>
<span class="co">## rowData names(9): Accession Gene ... genus_species isgene</span>
<span class="co">## colnames: NULL</span>
<span class="co">## colData names(44): CellID analysis_protocol.protocol_core.protocol_id</span>
<span class="co">##   ... specimen_from_organism.provenance.document_id total_umis</span>
<span class="co">## reducedDimNames(0):</span>
<span class="co">## altExpNames(0):</span></code></pre></div>
<ul>
<li>…ready for down-stream exploration and analysis.</li>
</ul>
</div>
<div id="writing--loom-files" class="section level1">
<h1 class="hasAnchor">
<a href="#writing--loom-files" class="anchor"></a>Writing .loom files</h1>
<p>Use <code><a href="https://rdrr.io/pkg/LoomExperiment/man/export-method.html">export()</a></code> to create a loom file for sharing with others.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">my_loom</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html">tempfile</a></span><span class="op">(</span>fileext <span class="op">=</span><span class="st">".loom"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/LoomExperiment/man/export-method.html">export</a></span><span class="op">(</span><span class="fu">as</span><span class="op">(</span><span class="va">sce</span>, <span class="st">"LoomExperiment"</span><span class="op">)</span>, <span class="va">my_loom</span><span class="op">)</span></code></pre></div>
</div>
<div id="summary" class="section level1">
<h1 class="hasAnchor">
<a href="#summary" class="anchor"></a>Summary</h1>
<ul>
<li>Pre-computed data matrices are easy to discover from <a href="https://data.humancellatlas.org/explore">HCA Data Portal</a>
</li>
<li>
<code>.loom</code> format provides very rich row (gene) and column (cell) annotations.</li>
<li>
<a href="https://bioconductor.org/packages/LoomExperiment">LoomExperiment</a> imports data into R in a memory-efficient manner</li>
<li>It is easy to integrate LoomExperiment objects into the rich R / Bioconductor infrastructure for downstream analysis.</li>
</ul>
<div id="session-info" class="section level2">
<h2 class="hasAnchor">
<a href="#session-info" class="anchor"></a>Session info</h2>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html">sessionInfo</a></span><span class="op">(</span><span class="op">)</span>
<span class="co">## R version 4.0.3 Patched (2020-10-13 r79345)</span>
<span class="co">## Platform: x86_64-apple-darwin17.7.0 (64-bit)</span>
<span class="co">## Running under: macOS High Sierra 10.13.6</span>
<span class="co">## </span>
<span class="co">## Matrix products: default</span>
<span class="co">## BLAS:   /Users/ma38727/bin/R-4-0-branch/lib/libRblas.dylib</span>
<span class="co">## LAPACK: /Users/ma38727/bin/R-4-0-branch/lib/libRlapack.dylib</span>
<span class="co">## </span>
<span class="co">## locale:</span>
<span class="co">## [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8</span>
<span class="co">## </span>
<span class="co">## attached base packages:</span>
<span class="co">## [1] stats4    parallel  stats     graphics  grDevices utils     datasets </span>
<span class="co">## [8] methods   base     </span>
<span class="co">## </span>
<span class="co">## other attached packages:</span>
<span class="co">##  [1] HCAMatrixBrowser_1.0.1      AnVIL_1.2.0                </span>
<span class="co">##  [3] dplyr_1.0.2                 LoomExperiment_1.8.0       </span>
<span class="co">##  [5] rtracklayer_1.50.0          rhdf5_2.34.0               </span>
<span class="co">##  [7] SingleCellExperiment_1.12.0 SummarizedExperiment_1.20.0</span>
<span class="co">##  [9] Biobase_2.50.0              GenomicRanges_1.42.0       </span>
<span class="co">## [11] GenomeInfoDb_1.26.0         IRanges_2.24.0             </span>
<span class="co">## [13] S4Vectors_0.28.0            BiocGenerics_0.36.0        </span>
<span class="co">## [15] MatrixGenerics_1.2.0        matrixStats_0.57.0         </span>
<span class="co">## </span>
<span class="co">## loaded via a namespace (and not attached):</span>
<span class="co">##  [1] httr_1.4.2               bit64_4.0.5              jsonlite_1.7.1          </span>
<span class="co">##  [4] assertthat_0.2.1         BiocFileCache_1.14.0     blob_1.2.1              </span>
<span class="co">##  [7] GenomeInfoDbData_1.2.4   Rsamtools_2.6.0          yaml_2.2.1              </span>
<span class="co">## [10] pillar_1.4.6             RSQLite_2.2.1            backports_1.2.0         </span>
<span class="co">## [13] lattice_0.20-41          glue_1.4.2               digest_0.6.27           </span>
<span class="co">## [16] XVector_0.30.0           htmltools_0.5.0          Matrix_1.2-18           </span>
<span class="co">## [19] XML_3.99-0.5             pkgconfig_2.0.3          zlibbioc_1.36.0         </span>
<span class="co">## [22] purrr_0.3.4              rapiclient_0.1.3         HDF5Array_1.18.0        </span>
<span class="co">## [25] BiocParallel_1.24.0      tibble_3.0.4             generics_0.1.0          </span>
<span class="co">## [28] ellipsis_0.3.1           withr_2.3.0              cli_2.1.0               </span>
<span class="co">## [31] magrittr_1.5             crayon_1.3.4             memoise_1.1.0           </span>
<span class="co">## [34] evaluate_0.14            fansi_0.4.1              fs_1.5.0                </span>
<span class="co">## [37] textshaping_0.1.2        tools_4.0.3              formatR_1.7             </span>
<span class="co">## [40] lifecycle_0.2.0          stringr_1.4.0            Rhdf5lib_1.12.0         </span>
<span class="co">## [43] DelayedArray_0.16.0      lambda.r_1.2.4           Biostrings_2.58.0       </span>
<span class="co">## [46] compiler_4.0.3           pkgdown_1.6.1            systemfonts_0.3.2       </span>
<span class="co">## [49] rlang_0.4.8              futile.logger_1.4.3      grid_4.0.3              </span>
<span class="co">## [52] RCurl_1.98-1.2           rhdf5filters_1.2.0       rappdirs_0.3.1          </span>
<span class="co">## [55] bitops_1.0-6             rmarkdown_2.5            codetools_0.2-18        </span>
<span class="co">## [58] DBI_1.1.0                curl_4.3                 R6_2.5.0                </span>
<span class="co">## [61] GenomicAlignments_1.26.0 knitr_1.30               utf8_1.1.4              </span>
<span class="co">## [64] bit_4.0.4                rprojroot_1.3-2          ragg_0.4.0              </span>
<span class="co">## [67] futile.options_1.0.1     desc_1.2.0               stringi_1.5.3           </span>
<span class="co">## [70] Rcpp_1.0.5               vctrs_0.3.4              dbplyr_2.0.0            </span>
<span class="co">## [73] tidyselect_1.1.0         xfun_0.19</span></code></pre></div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Martin Morgan.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
