<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Programmatic exploration of HCA data • BiocHCADataAccess</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Programmatic exploration of HCA data">
<meta property="og:description" content="BiocHCADataAccess">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">BiocHCADataAccess</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.0.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/A_SingleCellDataRepresentation.html">Representing single cell data in R / Bioconductor</a>
    </li>
    <li>
      <a href="../articles/B_HCADataDiscovery.html">HCA Data Discovery</a>
    </li>
    <li>
      <a href="../articles/C_HCAMatrixBrowser.html">Programmatic exploration of HCA data</a>
    </li>
    <li>
      <a href="../articles/D_HCAExplorer.html">Even more in-depth data discovery</a>
    </li>
    <li>
      <a href="../articles/E_Directions.html">Directions</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/mtmorgan/BiocHCADataAccess">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="C_HCAMatrixBrowser_files/header-attrs-2.5/header-attrs.js"></script><link href="C_HCAMatrixBrowser_files/anchor-sections-1.0/anchor-sections.css" rel="stylesheet">
<script src="C_HCAMatrixBrowser_files/anchor-sections-1.0/anchor-sections.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Programmatic exploration of HCA data</h1>
            
      
      
      <div class="hidden name"><code>C_HCAMatrixBrowser.Rmd</code></div>

    </div>

    
    
<p>Objectives:</p>
<ul>
<li>Explore HCA matrix data programmatically</li>
<li>Use the <a href="https://bioconductor.org/packages/HCAMatrixBrowser">HCAMatrixBrowser</a> ‘high level’ interface to filter, create, and retrieve a <a href="https://bioconductor.org/packages/LoomExperiment">LoomExperiment</a> with a custom count matrix.</li>
<li>Introduce <a href="https://bioconductor.org/packages/HCAMatrixBrowser">HCAMatrixBrowser</a> ‘low level’ operations for matrix retrieval.</li>
</ul>
<div id="the-hcamatrixbrowser-package" class="section level1">
<h1 class="hasAnchor">
<a href="#the-hcamatrixbrowser-package" class="anchor"></a>The HCAMatrixBrowser package</h1>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">HCAMatrixBrowser</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/utils/citation.html">citation</a></span><span class="op">(</span><span class="st">"HCAMatrixBrowser"</span><span class="op">)</span>, style <span class="op">=</span> <span class="st">"text"</span><span class="op">)</span>
<span class="co">## Ramos M, Morgan M (2020). _HCAMatrixBrowser: Extract and manage matrix</span>
<span class="co">## data from the Human Cell Atlas project_. R package version 1.0.1.</span></code></pre></div>
<p>Built on the <a href="https://matrix.data.humancellatlas.org/">HCA Matrix Service</a> RESTful API.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">hca</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/HCAMatrixBrowser/man/HCAMatrix.html">HCAMatrix</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<ul>
<li>Create an object that references the service.</li>
</ul>
</div>
<div id="high-level-use" class="section level1">
<h1 class="hasAnchor">
<a href="#high-level-use" class="anchor"></a>‘High-level’ use</h1>
<p>Discover available filters.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">filters</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/HCAMatrixBrowser/man/HCAMatrix.html">available_filters</a></span><span class="op">(</span><span class="va">hca</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">filters</span><span class="op">)</span>
<span class="co">## [1] 43</span>
<span class="fu"><a href="https://rdrr.io/r/base/grep.html">grep</a></span><span class="op">(</span><span class="st">"project"</span>, <span class="va">filters</span>, value <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>
<span class="co">## [1] "project.provenance.document_id"         </span>
<span class="co">## [2] "project.project_core.project_short_name"</span>
<span class="co">## [3] "project.project_core.project_title"</span></code></pre></div>
<ul>
<li>Filters available on 43 different facets</li>
<li>The <code>project.project_core.project_title</code> filter corresponds to titles on the matrix browser web page.</li>
</ul>
<p>Select a particular experiment</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">title</span> <span class="op">&lt;-</span> <span class="st">"Assessing the relevance of organoids to model inter-individual variation"</span>
<span class="va">hca1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/HCAMatrixBrowser/man/filtering.html">filter</a></span><span class="op">(</span>
    <span class="va">hca</span>,
    <span class="va">project.project_core.project_title</span> <span class="op">==</span> <span class="va">title</span>
<span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/HCAMatrixBrowser/man/filtering.html">filters</a></span><span class="op">(</span><span class="va">hca1</span><span class="op">)</span> <span class="op">%&gt;%</span>
    <span class="fu"><a href="https://rdrr.io/r/utils/str.html">str</a></span><span class="op">(</span><span class="op">)</span>
<span class="co">## List of 3</span>
<span class="co">##  $ op   : 'scalar' chr "="</span>
<span class="co">##  $ field: chr "project.project_core.project_title"</span>
<span class="co">##  $ value: chr "Assessing the relevance of organoids to model inter-individual variation"</span></code></pre></div>
<p>Combine filters</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/grep.html">grep</a></span><span class="op">(</span><span class="st">"detected"</span>, <span class="va">filters</span>, value <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>
<span class="co">## [1] "genes_detected"</span>
<span class="fu"><a href="https://rdrr.io/pkg/HCAMatrixBrowser/man/HCAMatrix.html">filter_detail</a></span><span class="op">(</span><span class="va">hca</span>, <span class="st">"genes_detected"</span><span class="op">)</span> <span class="op">%&gt;%</span>
    <span class="fu"><a href="https://rdrr.io/r/utils/str.html">str</a></span><span class="op">(</span><span class="op">)</span>
<span class="co">## List of 5</span>
<span class="co">##  $ field_description: chr "Count of genes with a non-zero count."</span>
<span class="co">##  $ field_name       : chr "genes_detected"</span>
<span class="co">##  $ field_type       : chr "numeric"</span>
<span class="co">##  $ maximum          : int 13108</span>
<span class="co">##  $ minimum          : int 2</span></code></pre></div>
<ul>
<li>
<code>genes_detected</code> can be used to filter only cells with more than 500 genes observed.</li>
<li>Combine with project selection.</li>
</ul>
<pre><code>hca1 &lt;- filter(
    hca,
    project.project_core.project_title == title &amp;
    genes_detected &gt;= 500
)
filters(hca1)</code></pre>
</div>
<div id="retrieve-the-custom-selection-of-cells" class="section level1">
<h1 class="hasAnchor">
<a href="#retrieve-the-custom-selection-of-cells" class="anchor"></a>Retrieve the custom selection of cells</h1>
<pre><code>&gt; loom &lt;- loadHCAMatrix(hca1, format = "loom")
HCAMatrixBrowser cache directory set to:
    /Users/ma38727/Library/Caches/org.R-project.R/R/HCAMatrixBrowser
  (\) waiting for query completion.......... 00:00:02
  (-) waiting for query completion.......... 00:00:06
...
  (/) waiting for query completion.......... 00:08:13
  |======================================================================| 100%</code></pre>
<ul>
<li>Matrix assembled by the service (remotely) ‘on the fly’</li>
</ul>
<pre><code>&gt; loom
class: LoomExperiment 
dim: 58347 43861 
metadata(0):
assays(1): matrix
rownames: NULL
rowData names(9): Accession Gene ... genus_species isgene
colnames(43861): 0006d1a2669431b298d08f010da6480b
  000f429f8f0c461ce9fe5a82b0375994 ... ffeb17d412dc2de0ac6e421f85eb15db
  ffef9246733b290086fd66e084905567
colData names(44): CellID analysis_protocol.protocol_core.protocol_id
  ... specimen_from_organism.provenance.document_id total_umis
rowGraphs(0): NULL
colGraphs(0): NULL
&gt; range(colSums(assay(loom)[, 1:1000] &gt; 0))
[1] 502 4660</code></pre>
<ul>
<li>The size of the experiment, compared to full version retrieved earlier, and the range of genes per cell give us confidence that the filters have been applied correctly.</li>
</ul>
</div>
<div id="low-level-operations" class="section level1">
<h1 class="hasAnchor">
<a href="#low-level-operations" class="anchor"></a>‘Low-level’ operations</h1>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">hca</span> <span class="op">%&gt;%</span>
    <span class="fu">tags</span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span>
    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span><span class="op">(</span><span class="va">tag</span> <span class="op">==</span> <span class="st">"v1"</span><span class="op">)</span> <span class="op">%&gt;%</span>
    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span><span class="op">(</span><span class="va">operation</span><span class="op">)</span>
<span class="co">## # A tibble: 10 x 1</span>
<span class="co">##    operation                                    </span>
<span class="co">##    &lt;chr&gt;                                        </span>
<span class="co">##  1 matrix.lambdas.api.v1.core.get_feature_detail</span>
<span class="co">##  2 matrix.lambdas.api.v1.core.get_features      </span>
<span class="co">##  3 matrix.lambdas.api.v1.core.get_field_detail  </span>
<span class="co">##  4 matrix.lambdas.api.v1.core.get_fields        </span>
<span class="co">##  5 matrix.lambdas.api.v1.core.get_filter_detail </span>
<span class="co">##  6 matrix.lambdas.api.v1.core.get_filters       </span>
<span class="co">##  7 matrix.lambdas.api.v1.core.get_format_detail </span>
<span class="co">##  8 matrix.lambdas.api.v1.core.get_formats       </span>
<span class="co">##  9 matrix.lambdas.api.v1.core.get_matrix        </span>
<span class="co">## 10 matrix.lambdas.api.v1.core.post_matrix</span>

<span class="va">hca</span><span class="op">$</span><span class="va">matrix.lambdas.api.v1.core.post_matrix</span>
<span class="co">## matrix.lambdas.api.v1.core.post_matrix </span>
<span class="co">## matrix.lambdas.api.v1.core.post_matrix </span>
<span class="co">## Description:</span>
<span class="co">##   Request an expression matrix using a set of filters to be ANDed and</span>
<span class="co">##   applied to the HCA expression data.</span>
<span class="co">## </span>
<span class="co">## Parameters:</span>
<span class="co">##   filter (object)</span>
<span class="co">##     Filter to apply in a matrix request.</span>
<span class="co">##   fields (array[string])</span>
<span class="co">##     Metadata fields to include in output matrix.</span>
<span class="co">##   format (string)</span>
<span class="co">##     Format for the output matrix.</span>
<span class="co">##   feature (string)</span>
<span class="co">##     Feature type to include in the output matrix. For example, genes</span>
<span class="co">##     vs. transcripts.</span></code></pre></div>
<ul>
<li>End-points exposed to the R user.</li>
<li>
<code>matrix.lambdas.api.v1.core.post_matrix()</code> requests construction of a customized count matrix, based on a series of filters, metadata fields, etc.</li>
</ul>
</div>
<div id="summary" class="section level1">
<h1 class="hasAnchor">
<a href="#summary" class="anchor"></a>Summary</h1>
<ul>
<li>
<a href="https://bioconductor.org/packages/HCAMatrixBrowser">HCAMatrixBrowser</a> allows programmatic access to the HCA Matrix Service.</li>
<li>Both complete and data filtered to precise criteria can be retrieved.</li>
<li>The full RESTful HCA Matrix Service API is available for further customization.</li>
</ul>
<div id="session-info" class="section level2">
<h2 class="hasAnchor">
<a href="#session-info" class="anchor"></a>Session info</h2>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html">sessionInfo</a></span><span class="op">(</span><span class="op">)</span>
<span class="co">## R version 4.0.3 Patched (2020-10-13 r79345)</span>
<span class="co">## Platform: x86_64-apple-darwin17.7.0 (64-bit)</span>
<span class="co">## Running under: macOS High Sierra 10.13.6</span>
<span class="co">## </span>
<span class="co">## Matrix products: default</span>
<span class="co">## BLAS:   /Users/ma38727/bin/R-4-0-branch/lib/libRblas.dylib</span>
<span class="co">## LAPACK: /Users/ma38727/bin/R-4-0-branch/lib/libRlapack.dylib</span>
<span class="co">## </span>
<span class="co">## locale:</span>
<span class="co">## [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8</span>
<span class="co">## </span>
<span class="co">## attached base packages:</span>
<span class="co">## [1] stats     graphics  grDevices utils     datasets  methods   base     </span>
<span class="co">## </span>
<span class="co">## other attached packages:</span>
<span class="co">## [1] HCAMatrixBrowser_1.0.1 AnVIL_1.2.0            dplyr_1.0.2           </span>
<span class="co">## </span>
<span class="co">## loaded via a namespace (and not attached):</span>
<span class="co">##  [1] Rcpp_1.0.5                  lattice_0.20-41            </span>
<span class="co">##  [3] utf8_1.1.4                  assertthat_0.2.1           </span>
<span class="co">##  [5] rprojroot_1.3-2             digest_0.6.27              </span>
<span class="co">##  [7] SingleCellExperiment_1.12.0 BiocFileCache_1.14.0       </span>
<span class="co">##  [9] R6_2.5.0                    GenomeInfoDb_1.26.0        </span>
<span class="co">## [11] futile.options_1.0.1        backports_1.2.0            </span>
<span class="co">## [13] rapiclient_0.1.3            stats4_4.0.3               </span>
<span class="co">## [15] RSQLite_2.2.1               evaluate_0.14              </span>
<span class="co">## [17] httr_1.4.2                  pillar_1.4.6               </span>
<span class="co">## [19] zlibbioc_1.36.0             rlang_0.4.8                </span>
<span class="co">## [21] curl_4.3                    blob_1.2.1                 </span>
<span class="co">## [23] S4Vectors_0.28.0            Matrix_1.2-18              </span>
<span class="co">## [25] rmarkdown_2.5               pkgdown_1.6.1              </span>
<span class="co">## [27] textshaping_0.1.2           desc_1.2.0                 </span>
<span class="co">## [29] stringr_1.4.0               RCurl_1.98-1.2             </span>
<span class="co">## [31] bit_4.0.4                   DelayedArray_0.16.0        </span>
<span class="co">## [33] compiler_4.0.3              xfun_0.19                  </span>
<span class="co">## [35] pkgconfig_2.0.3             systemfonts_0.3.2          </span>
<span class="co">## [37] BiocGenerics_0.36.0         htmltools_0.5.0            </span>
<span class="co">## [39] tidyselect_1.1.0            SummarizedExperiment_1.20.0</span>
<span class="co">## [41] tibble_3.0.4                GenomeInfoDbData_1.2.4     </span>
<span class="co">## [43] codetools_0.2-18            IRanges_2.24.0             </span>
<span class="co">## [45] matrixStats_0.57.0          fansi_0.4.1                </span>
<span class="co">## [47] crayon_1.3.4                dbplyr_2.0.0               </span>
<span class="co">## [49] bitops_1.0-6                rappdirs_0.3.1             </span>
<span class="co">## [51] grid_4.0.3                  jsonlite_1.7.1             </span>
<span class="co">## [53] lifecycle_0.2.0             DBI_1.1.0                  </span>
<span class="co">## [55] magrittr_1.5                formatR_1.7                </span>
<span class="co">## [57] cli_2.1.0                   stringi_1.5.3              </span>
<span class="co">## [59] XVector_0.30.0              fs_1.5.0                   </span>
<span class="co">## [61] futile.logger_1.4.3         ellipsis_0.3.1             </span>
<span class="co">## [63] ragg_0.4.0                  generics_0.1.0             </span>
<span class="co">## [65] vctrs_0.3.4                 lambda.r_1.2.4             </span>
<span class="co">## [67] tools_4.0.3                 bit64_4.0.5                </span>
<span class="co">## [69] Biobase_2.50.0              glue_1.4.2                 </span>
<span class="co">## [71] purrr_0.3.4                 MatrixGenerics_1.2.0       </span>
<span class="co">## [73] parallel_4.0.3              yaml_2.2.1                 </span>
<span class="co">## [75] GenomicRanges_1.42.0        memoise_1.1.0              </span>
<span class="co">## [77] knitr_1.30</span></code></pre></div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Martin Morgan.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
