<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>In-depth data discovery • BiocHCADataAccess</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="In-depth data discovery">
<meta property="og:description" content="BiocHCADataAccess">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">BiocHCADataAccess</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.0.3</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/A_SingleCellDataRepresentation.html">Representing single cell data in R / Bioconductor</a>
    </li>
    <li>
      <a href="../articles/B_HCADataDiscovery.html">HCA Data Discovery</a>
    </li>
    <li>
      <a href="../articles/C_HCAMatrixBrowser.html">Programmatic exploration of HCA data</a>
    </li>
    <li>
      <a href="../articles/D_HCAExplorer.html">In-depth data discovery</a>
    </li>
    <li>
      <a href="../articles/E_Directions.html">Directions and Acknowledgments</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/mtmorgan/BiocHCADataAccess">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="D_HCAExplorer_files/header-attrs-2.5/header-attrs.js"></script><link href="D_HCAExplorer_files/anchor-sections-1.0/anchor-sections.css" rel="stylesheet">
<script src="D_HCAExplorer_files/anchor-sections-1.0/anchor-sections.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>In-depth data discovery</h1>
            
      
      
      <div class="hidden name"><code>D_HCAExplorer.Rmd</code></div>

    </div>

    
    
<p>Objectives:</p>
<ul>
<li>Use the <a href="https://bioconductor.org/packages/HCAExplorer">HCAExplorer</a> package to identify fastq files for primary analysis.</li>
</ul>
<div id="the-hcaexplorer-package" class="section level1">
<h1 class="hasAnchor">
<a href="#the-hcaexplorer-package" class="anchor"></a>The HCAExplorer package</h1>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/Bioconductor/HCABrowser">HCAExplorer</a></span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/utils/citation.html">citation</a></span><span class="op">(</span><span class="st">"HCAExplorer"</span><span class="op">)</span>, style <span class="op">=</span> <span class="st">"text"</span><span class="op">)</span>
<span class="co">## Van Twisk D, Morgan M (2020). _HCAExplorer: Browse the Human Cell Atlas</span>
<span class="co">## data portal_. R package version 1.4.0, &lt;URL:</span>
<span class="co">## https://github.com/Bioconductor/HCABrowser&gt;.</span></code></pre></div>
<p>Built on the <a href="https://service.explore.data.humancellatlas.org">HCA azul</a> RESTful API.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">hca</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/HCAExplorer/man/HCAExplorer.html">HCAExplorer</a></span><span class="op">(</span><span class="op">)</span>
<span class="va">hca</span>
<span class="co">## class: HCAExplorer </span>
<span class="co">## Using azul backend at:</span>
<span class="co">##   https://service.explore.data.humancellatlas.org </span>
<span class="co">## </span>
<span class="co">## Donor count: 289 </span>
<span class="co">## Specimens: 780 </span>
<span class="co">## Estimated Cells: 4535231 </span>
<span class="co">## Files: 536961 </span>
<span class="co">## File Size: 30.2 Tb </span>
<span class="co">## </span>
<span class="co">## Showing projects with 15 results per page.# A tibble: 15 x 7</span>
<span class="co">##    projects.projec… samples.sampleE… samples.organ protocols.libra…</span>
<span class="co">##    &lt;chr&gt;            &lt;chr&gt;            &lt;chr&gt;         &lt;chr&gt;           </span>
<span class="co">##  1 1.3 Million Bra… specimens        brain         10X v2 sequenci…</span>
<span class="co">##  2 A Single-Cell T… specimens        pancreas      inDrop, NULL    </span>
<span class="co">##  3 A single-cell m… specimens        embryo        10X 3' v1 seque…</span>
<span class="co">##  4 A single-cell r… specimens        blood, hemat… 10X v2 sequenci…</span>
<span class="co">##  5 A single-cell t… specimens        eye           10X v2 sequenci…</span>
<span class="co">##  6 Assessing the r… organoids        &lt;NA&gt;          10X v2 sequenci…</span>
<span class="co">##  7 Bone marrow pla… specimens        hematopoieti… MARS-seq, NULL  </span>
<span class="co">##  8 Cell hashing wi… specimens        blood         CITE-seq, NULL  </span>
<span class="co">##  9 Census of Immun… specimens        blood, immun… 10X v2 sequenci…</span>
<span class="co">## 10 Comparison, cal… cellLines        &lt;NA&gt;          DroNc-Seq, Drop…</span>
<span class="co">## 11 Dissecting the … specimens        liver         10X v2 sequenci…</span>
<span class="co">## 12 Ischaemic sensi… specimens        esophagus, l… 10X v2 sequenci…</span>
<span class="co">## 13 Melanoma infilt… specimens        lymph node, … Smart-seq2, NULL</span>
<span class="co">## 14 Precursors of h… specimens        blood         Smart-seq2, NULL</span>
<span class="co">## 15 Profiling of CD… cellLines        &lt;NA&gt;          10X v2 sequenci…</span>
<span class="co">## # … with 3 more variables: protocols.pairedEnd &lt;chr&gt;,</span>
<span class="co">## #   donorOrganisms.genusSpecies &lt;chr&gt;, samples.disease &lt;chr&gt;</span>
<span class="co">## Showing page 1 of 2</span></code></pre></div>
</div>
<div id="filtering-to-a-specific-project" class="section level1">
<h1 class="hasAnchor">
<a href="#filtering-to-a-specific-project" class="anchor"></a>Filtering to a specific project</h1>
<p>Identify our favorite project and retrieve detailed metadata about it.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">title</span> <span class="op">&lt;-</span> <span class="st">"Assessing the relevance of organoids to model inter-individual variation"</span>
<span class="va">hca</span> <span class="op">%&gt;%</span>
    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span><span class="op">(</span><span class="va">projectTitle</span> <span class="op">%in%</span> <span class="va">title</span><span class="op">)</span> <span class="op">%&gt;%</span>
    <span class="fu"><a href="https://rdrr.io/pkg/HCAExplorer/man/results.html">results</a></span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span>
    <span class="fu"><a href="https://rdrr.io/r/utils/str.html">str</a></span><span class="op">(</span><span class="op">)</span>
<span class="co">## tibble [1 × 46] (S3: tbl_df/tbl/data.frame)</span>
<span class="co">##  $ protocols.libraryConstructionApproach: chr "10X v2 sequencing, NULL"</span>
<span class="co">##  $ protocols.instrumentManufacturerModel: chr "Illumina HiSeq 2500, NULL"</span>
<span class="co">##  $ protocols.pairedEnd                  : chr "TRUE, NULL"</span>
<span class="co">##  $ protocols.workflow                   : chr "optimus_v1.3.1, optimus_v1.3.2, NULL"</span>
<span class="co">##  $ protocols.assayType                  : chr "NULL"</span>
<span class="co">##  $ entryId                              : chr "005d611a-14d5-4fbf-846e-571a1f874f70"</span>
<span class="co">##  $ projects.projectTitle                : chr "Assessing the relevance of organoids to model inter-individual variation"</span>
<span class="co">##  $ projects.projectShortname            : chr "HPSI human cerebral organoids"</span>
<span class="co">##  $ projects.laboratory                  : chr "Human Cell Atlas Data Coordination Platform"</span>
<span class="co">##  $ projects.arrayExpressAccessions      : chr "NULL"</span>
<span class="co">##  $ projects.geoSeriesAccessions         : chr "NULL"</span>
<span class="co">##  $ projects.insdcProjectAccessions      : chr "ERP114427"</span>
<span class="co">##  $ projects.insdcStudyAccessions        : chr "PRJEB31821"</span>
<span class="co">##  $ projects.supplementaryLinks          : chr "NULL"</span>
<span class="co">##  $ samples.sampleEntityType             : chr "organoids"</span>
<span class="co">##  $ samples.effectiveOrgan               : chr "brain"</span>
<span class="co">##  $ samples.modelOrganPart               : chr "NULL"</span>
<span class="co">##  $ samples.id                           : chr "Org_HPSI0214i-kucg_2_1, Org_HPSI0214i-kucg_2_2, Org_HPSI0214i-kucg_2_3, Org_HPSI0214i-wibj_2_1, Org_HPSI0214i-w"| __truncated__</span>
<span class="co">##  $ samples.modelOrgan                   : chr "brain"</span>
<span class="co">##  $ specimens.id                         : chr "HPSI0214i-kucg_skin, HPSI0214i-wibj_skin, HPSI0314i-hoik_skin, HPSI0314i-sojd_skin"</span>
<span class="co">##  $ specimens.organ                      : chr "skin of body"</span>
<span class="co">##  $ specimens.organPart                  : chr "skin epidermis"</span>
<span class="co">##  $ specimens.disease                    : chr "normal"</span>
<span class="co">##  $ specimens.preservationMethod         : chr "NULL"</span>
<span class="co">##  $ specimens.source                     : chr "specimen_from_organism"</span>
<span class="co">##  $ cellLines.id                         : chr "HPSI0214i-kucg_2, HPSI0214i-wibj_2, HPSI0314i-hoik_1, HPSI0314i-sojd_3"</span>
<span class="co">##  $ cellLines.cellLineType               : chr "induced pluripotent"</span>
<span class="co">##  $ cellLines.modelOrgan                 : chr "stem cell"</span>
<span class="co">##  $ donorOrganisms.id                    : chr "HPSI0214i-kucg, HPSI0214i-wibj, HPSI0314i-hoik, HPSI0314i-sojd"</span>
<span class="co">##  $ donorOrganisms.donorCount            : chr "4"</span>
<span class="co">##  $ donorOrganisms.genusSpecies          : chr "Homo sapiens"</span>
<span class="co">##  $ donorOrganisms.organismAge           : chr "40-44, 45-49, 55-59, 65-69"</span>
<span class="co">##  $ donorOrganisms.organismAgeUnit       : chr "year"</span>
<span class="co">##  $ donorOrganisms.organismAgeRange      : chr "list(gte = 1261440000, lte = 1387584000), list(gte = 1419120000, lte = 1545264000), list(gte = 1734480000, lte "| __truncated__</span>
<span class="co">##  $ donorOrganisms.biologicalSex         : chr "female, male"</span>
<span class="co">##  $ donorOrganisms.disease               : chr "normal"</span>
<span class="co">##  $ organoids.id                         : chr "Org_HPSI0214i-kucg_2_1, Org_HPSI0214i-kucg_2_2, Org_HPSI0214i-kucg_2_3, Org_HPSI0214i-wibj_2_1, Org_HPSI0214i-w"| __truncated__</span>
<span class="co">##  $ organoids.modelOrgan                 : chr "brain"</span>
<span class="co">##  $ organoids.modelOrganPart             : chr "NULL"</span>
<span class="co">##  $ cellSuspensions.organ                : chr "brain"</span>
<span class="co">##  $ cellSuspensions.organPart            : chr "NULL"</span>
<span class="co">##  $ cellSuspensions.selectedCellType     : chr "neural cell"</span>
<span class="co">##  $ cellSuspensions.totalCells           : chr "19916"</span>
<span class="co">##  $ fileTypeSummaries.fileType           : chr "fastq.gz"</span>
<span class="co">##  $ fileTypeSummaries.count              : chr "18"</span>
<span class="co">##  $ fileTypeSummaries.totalSize          : chr "53857782294"</span></code></pre></div>
</div>
<div id="obtaining-a-manifest-of-fastq-files-" class="section level1">
<h1 class="hasAnchor">
<a href="#obtaining-a-manifest-of-fastq-files-" class="anchor"></a>Obtaining a ‘manifest’ of fastq files.</h1>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">manifest</span> <span class="op">&lt;-</span>
    <span class="va">hca</span> <span class="op">%&gt;%</span>
    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span><span class="op">(</span><span class="va">projectTitle</span> <span class="op">%in%</span> <span class="va">title</span><span class="op">)</span> <span class="op">%&gt;%</span>
    <span class="fu"><a href="https://rdrr.io/pkg/HCAExplorer/man/getManifest.html">getManifest</a></span><span class="op">(</span>fileFormat <span class="op">=</span> <span class="st">"fastq.gz"</span><span class="op">)</span>
<span class="co">## sleeping 1 seconds</span>
<span class="co">## sleeping 4 seconds</span>
<span class="co">## No encoding supplied: defaulting to UTF-8.</span>
<span class="co">## </span>
<span class="co">## ── Column specification ────────────────────────────────────────────────────────</span>
<span class="co">## cols(</span>
<span class="co">##   .default = col_character(),</span>
<span class="co">##   bundle_version = col_datetime(format = ""),</span>
<span class="co">##   file_size = col_double(),</span>
<span class="co">##   file_version = col_datetime(format = ""),</span>
<span class="co">##   cell_suspension.estimated_cell_count = col_double(),</span>
<span class="co">##   sequencing_protocol.paired_end = col_logical(),</span>
<span class="co">##   specimen_from_organism.preservation_storage.preservation_method = col_logical(),</span>
<span class="co">##   organoid.model_organ_part = col_logical()</span>
<span class="co">## )</span>
<span class="co">## ℹ Use `spec()` for the full column specifications.</span></code></pre></div>
<ul>
<li>Manifest created by the service.</li>
</ul>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">manifest</span><span class="op">)</span>
<span class="co">##  [1] "bundle_uuid"                                                    </span>
<span class="co">##  [2] "bundle_version"                                                 </span>
<span class="co">##  [3] "file_name"                                                      </span>
<span class="co">##  [4] "file_format"                                                    </span>
<span class="co">##  [5] "read_index"                                                     </span>
<span class="co">##  [6] "file_size"                                                      </span>
<span class="co">##  [7] "file_uuid"                                                      </span>
<span class="co">##  [8] "file_version"                                                   </span>
<span class="co">##  [9] "file_sha256"                                                    </span>
<span class="co">## [10] "file_content_type"                                              </span>
<span class="co">## [11] "cell_suspension.provenance.document_id"                         </span>
<span class="co">## [12] "cell_suspension.biomaterial_core.biomaterial_id"                </span>
<span class="co">## [13] "cell_suspension.estimated_cell_count"                           </span>
<span class="co">## [14] "cell_suspension.selected_cell_type"                             </span>
<span class="co">## [15] "sequencing_process.provenance.document_id"                      </span>
<span class="co">## [16] "sequencing_protocol.instrument_manufacturer_model"              </span>
<span class="co">## [17] "sequencing_protocol.paired_end"                                 </span>
<span class="co">## [18] "library_preparation_protocol.library_construction_approach"     </span>
<span class="co">## [19] "project.provenance.document_id"                                 </span>
<span class="co">## [20] "project.contributors.institution"                               </span>
<span class="co">## [21] "project.contributors.laboratory"                                </span>
<span class="co">## [22] "project.project_core.project_short_name"                        </span>
<span class="co">## [23] "project.project_core.project_title"                             </span>
<span class="co">## [24] "specimen_from_organism.provenance.document_id"                  </span>
<span class="co">## [25] "specimen_from_organism.diseases"                                </span>
<span class="co">## [26] "specimen_from_organism.organ"                                   </span>
<span class="co">## [27] "specimen_from_organism.organ_part"                              </span>
<span class="co">## [28] "specimen_from_organism.preservation_storage.preservation_method"</span>
<span class="co">## [29] "donor_organism.sex"                                             </span>
<span class="co">## [30] "donor_organism.biomaterial_core.biomaterial_id"                 </span>
<span class="co">## [31] "donor_organism.provenance.document_id"                          </span>
<span class="co">## [32] "donor_organism.genus_species"                                   </span>
<span class="co">## [33] "donor_organism.diseases"                                        </span>
<span class="co">## [34] "donor_organism.organism_age"                                    </span>
<span class="co">## [35] "donor_organism.organism_age_unit"                               </span>
<span class="co">## [36] "cell_line.provenance.document_id"                               </span>
<span class="co">## [37] "cell_line.biomaterial_core.biomaterial_id"                      </span>
<span class="co">## [38] "organoid.provenance.document_id"                                </span>
<span class="co">## [39] "organoid.biomaterial_core.biomaterial_id"                       </span>
<span class="co">## [40] "organoid.model_organ"                                           </span>
<span class="co">## [41] "organoid.model_organ_part"                                      </span>
<span class="co">## [42] "_entity_type"                                                   </span>
<span class="co">## [43] "sample.provenance.document_id"                                  </span>
<span class="co">## [44] "sample.biomaterial_core.biomaterial_id"</span></code></pre></div>
<ul>
<li>Considerable information available about each file.</li>
</ul>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">manifest</span> <span class="op">%&gt;%</span>
    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span><span class="op">(</span><span class="va">file_name</span>, <span class="va">file_uuid</span><span class="op">)</span> <span class="op">%&gt;%</span>
    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/distinct.html">distinct</a></span><span class="op">(</span><span class="op">)</span>
<span class="co">## # A tibble: 18 x 2</span>
<span class="co">##    file_name                                  file_uuid                         </span>
<span class="co">##    &lt;chr&gt;                                      &lt;chr&gt;                             </span>
<span class="co">##  1 GAC027_hOrg_HipSci_1_S5_L008_R1_001.fastq… 9b93ead3-ee3a-47ea-a69a-0af6a316c…</span>
<span class="co">##  2 GAC027_hOrg_HipSci_1_S5_L008_R2_001.fastq… c231df7c-2cbd-4aba-bfed-3f067416c…</span>
<span class="co">##  3 GAC027_hOrg_HipSci_1_S5_L008_I1_001.fastq… 2e3516ac-a3e4-490a-bfda-06520ee22…</span>
<span class="co">##  4 GAC027_hOrg_HipSci_1_S5_L007_R1_001.fastq… c8d30c50-8a9f-4976-9d82-20021525b…</span>
<span class="co">##  5 GAC027_hOrg_HipSci_1_S5_L007_R2_001.fastq… a00e5d23-3ddb-4d55-85b5-82dfaf5cb…</span>
<span class="co">##  6 GAC027_hOrg_HipSci_3_S7_L007_I1_001.fastq… 2978f39c-708e-46a6-91d0-0b5813887…</span>
<span class="co">##  7 GAC027_hOrg_HipSci_1_S5_L007_I1_001.fastq… c8e31b6a-ff6c-4a93-80a3-45ac2700f…</span>
<span class="co">##  8 GAC027_hOrg_HipSci_2_S6_L007_R1_001.fastq… 0b3a1085-53ab-464b-a135-512a8b6f8…</span>
<span class="co">##  9 GAC027_hOrg_HipSci_2_S6_L007_I1_001.fastq… ba3c5206-ef33-4fc3-bbf4-23fb3a67c…</span>
<span class="co">## 10 GAC027_hOrg_HipSci_2_S6_L007_R2_001.fastq… 4fae3cc3-eeb5-42d3-8323-7ea5c3ff9…</span>
<span class="co">## 11 GAC027_hOrg_HipSci_2_S6_L008_R1_001.fastq… dd1cd137-5404-40c6-94e4-5b3b8b2b4…</span>
<span class="co">## 12 GAC027_hOrg_HipSci_2_S6_L008_I1_001.fastq… 44806e72-70ab-4a8c-a568-26e9cda42…</span>
<span class="co">## 13 GAC027_hOrg_HipSci_3_S7_L008_R1_001.fastq… 6b1ee7b3-11cf-4c14-947e-0a8ca7678…</span>
<span class="co">## 14 GAC027_hOrg_HipSci_2_S6_L008_R2_001.fastq… 42efa3d8-9c97-489d-916f-47abb7919…</span>
<span class="co">## 15 GAC027_hOrg_HipSci_3_S7_L007_R1_001.fastq… 37198fa9-7933-4438-8445-8f895e116…</span>
<span class="co">## 16 GAC027_hOrg_HipSci_3_S7_L007_R2_001.fastq… f8e6fe23-e571-45cd-8a0e-c215bd5c9…</span>
<span class="co">## 17 GAC027_hOrg_HipSci_3_S7_L008_R2_001.fastq… 9158d752-61a9-4307-9a57-560769586…</span>
<span class="co">## 18 GAC027_hOrg_HipSci_3_S7_L008_I1_001.fastq… 2b20ed1a-7bf9-4e4a-950e-f0bd8df96…</span></code></pre></div>
<ul>
<li>File names mapped to file UUIDs.</li>
<li>Files can be retrieved by UUID, e.g., using the HCA CLI (command line interface).</li>
</ul>
</div>
<div id="summary" class="section level1">
<h1 class="hasAnchor">
<a href="#summary" class="anchor"></a>Summary</h1>
<ul>
<li>
<a href="https://bioconductor.org/packages/HCAExplorer">HCAExplorer</a> provides more flexible, but also more complicated, ways to explore the data</li>
<li>Enables tasks like identifying and downloading all FASTQ files associated with a project.</li>
</ul>
<div id="session-info" class="section level2">
<h2 class="hasAnchor">
<a href="#session-info" class="anchor"></a>Session info</h2>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html">sessionInfo</a></span><span class="op">(</span><span class="op">)</span>
<span class="co">## R version 4.0.3 Patched (2020-10-13 r79345)</span>
<span class="co">## Platform: x86_64-apple-darwin17.7.0 (64-bit)</span>
<span class="co">## Running under: macOS High Sierra 10.13.6</span>
<span class="co">## </span>
<span class="co">## Matrix products: default</span>
<span class="co">## BLAS:   /Users/ma38727/bin/R-4-0-branch/lib/libRblas.dylib</span>
<span class="co">## LAPACK: /Users/ma38727/bin/R-4-0-branch/lib/libRlapack.dylib</span>
<span class="co">## </span>
<span class="co">## locale:</span>
<span class="co">## [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8</span>
<span class="co">## </span>
<span class="co">## attached base packages:</span>
<span class="co">## [1] stats     graphics  grDevices utils     datasets  methods   base     </span>
<span class="co">## </span>
<span class="co">## other attached packages:</span>
<span class="co">## [1] HCAExplorer_1.4.0 dplyr_1.0.2      </span>
<span class="co">## </span>
<span class="co">## loaded via a namespace (and not attached):</span>
<span class="co">##  [1] MatrixGenerics_1.2.0        Biobase_2.50.0             </span>
<span class="co">##  [3] httr_1.4.2                  tidyr_1.1.2                </span>
<span class="co">##  [5] tidygraph_1.2.0             bit64_4.0.5                </span>
<span class="co">##  [7] jsonlite_1.7.1              assertthat_0.2.1           </span>
<span class="co">##  [9] stats4_4.0.3                BiocFileCache_1.14.0       </span>
<span class="co">## [11] blob_1.2.1                  GenomeInfoDbData_1.2.4     </span>
<span class="co">## [13] Rsamtools_2.6.0             yaml_2.2.1                 </span>
<span class="co">## [15] pillar_1.4.6                RSQLite_2.2.1              </span>
<span class="co">## [17] backports_1.2.0             lattice_0.20-41            </span>
<span class="co">## [19] glue_1.4.2                  digest_0.6.27              </span>
<span class="co">## [21] GenomicRanges_1.42.0        XVector_0.30.0             </span>
<span class="co">## [23] htmltools_0.5.0             Matrix_1.2-18              </span>
<span class="co">## [25] plyr_1.8.6                  XML_3.99-0.5               </span>
<span class="co">## [27] pkgconfig_2.0.3             zlibbioc_1.36.0            </span>
<span class="co">## [29] purrr_0.3.4                 HDF5Array_1.18.0           </span>
<span class="co">## [31] BiocParallel_1.24.0         tibble_3.0.4               </span>
<span class="co">## [33] generics_0.1.0              IRanges_2.24.0             </span>
<span class="co">## [35] ellipsis_0.3.1              SummarizedExperiment_1.20.0</span>
<span class="co">## [37] BiocGenerics_0.36.0         cli_2.1.0                  </span>
<span class="co">## [39] magrittr_1.5                crayon_1.3.4               </span>
<span class="co">## [41] ps_1.4.0                    memoise_1.1.0              </span>
<span class="co">## [43] evaluate_0.14               fansi_0.4.1                </span>
<span class="co">## [45] fs_1.5.0                    xml2_1.3.2                 </span>
<span class="co">## [47] textshaping_0.1.2           tools_4.0.3                </span>
<span class="co">## [49] hms_0.5.3                   lifecycle_0.2.0            </span>
<span class="co">## [51] matrixStats_0.57.0          stringr_1.4.0              </span>
<span class="co">## [53] Rhdf5lib_1.12.0             S4Vectors_0.28.0           </span>
<span class="co">## [55] DelayedArray_0.16.0         Biostrings_2.58.0          </span>
<span class="co">## [57] compiler_4.0.3              pkgdown_1.6.1              </span>
<span class="co">## [59] GenomeInfoDb_1.26.0         systemfonts_0.3.2          </span>
<span class="co">## [61] rlang_0.4.8                 rhdf5_2.34.0               </span>
<span class="co">## [63] grid_4.0.3                  RCurl_1.98-1.2             </span>
<span class="co">## [65] rstudioapi_0.11             rhdf5filters_1.2.0         </span>
<span class="co">## [67] LoomExperiment_1.8.0        rappdirs_0.3.1             </span>
<span class="co">## [69] SingleCellExperiment_1.12.0 igraph_1.2.6               </span>
<span class="co">## [71] bitops_1.0-6                rmarkdown_2.5              </span>
<span class="co">## [73] codetools_0.2-18            DBI_1.1.0                  </span>
<span class="co">## [75] curl_4.3                    R6_2.5.0                   </span>
<span class="co">## [77] GenomicAlignments_1.26.0    knitr_1.30                 </span>
<span class="co">## [79] rtracklayer_1.50.0          utf8_1.1.4                 </span>
<span class="co">## [81] bit_4.0.4                   rprojroot_1.3-2            </span>
<span class="co">## [83] ragg_0.4.0                  readr_1.4.0                </span>
<span class="co">## [85] desc_1.2.0                  stringi_1.5.3              </span>
<span class="co">## [87] parallel_4.0.3              Rcpp_1.0.5                 </span>
<span class="co">## [89] vctrs_0.3.4                 dbplyr_2.0.0               </span>
<span class="co">## [91] tidyselect_1.1.0            xfun_0.19</span></code></pre></div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Martin Morgan.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
